library/src/main/java/com/chuckerteam/chucker/internal/ui/MainViewModel.kt

```diff
@@ -35,6 +40,20 @@ internal class MainViewModel : ViewModel() {
     val throwables: LiveData<List<RecordedThrowableTuple>> = RepositoryProvider.throwable()
         .getSortedThrowablesTuples()
 
+    fun getAllTransactions(): List<HttpTransaction>? = runBlocking {
+        val transactions = async { RepositoryProvider.transaction().getAllTransactions() }
+        transactions.await()
+    }
+
+    fun createExportFile(content: String, fileFactory: FileFactory): File = runBlocking {
+        val file = async {
+            val file = fileFactory.create(fileFactory.exportFileName)
+            file.writeText(content)
+            return@async file
+        }
+        file.await()
+    }
```

Comment:
```suggestion
    fun createExportFile(content: String, fileFactory: FileFactory): File = runBlocking {
        val file = viewModelScope.async(Dispatchers.IO) {
            val file = fileFactory.create(EXPORT_FILENAME)
            file.writeText(content)
            return@async file
        }
        file.await()
    }
```

Same as above. Moreover you can maybe specify `Dispatchers.IO` here as you're doing IO work.

Reply:
updated.